docs/developer/file-structure.rst

```diff
@@ -75,112 +76,175 @@ omitted from this page in the interest of brevity.
+
+                     in their python file's list of imports. To the user, it
+                     would simply appear that ``anonymise_dataset`` is a
+                     function inside of a module called ``dicom`` that is
+                     contained in `pymedphys/src/pymedphys`. However, under the
+                     hood, ``anonymise_dataset`` is actually a function in
+                     a file called `anonymise.py` in level 2 of the
+                     `pymedphys_dicom` subpackage, or:
+
+                     *pymedphys/packages/pymedphys_dicom/src/pymedphys_dicom/dicom/_level2/anonymise.py*
+
+                     Thankfully, this long path is invisible to the user due to
+                     the imports included in the modules of *pymedphys* main.
+                     For more details on code structure and PyMedPhys'
+                     dependency handling, see :ref:`source-code`.
 
-:``tests/``: Contains the PyMedPhys suite of automated tests. Any code present
-             in *src/pymedphys/* should be covered by tests in this directory.
-             Automated testing is essential for effective `continuous
-             integration`_, which is a core development philosophy of
-             PyMedPhys. If you would like to make meaningful contributions to
-             PyMedPhys - and become a much better developer as a result - it
-             pays to get very familiar with automated testing and the code
-             within this directory.
 
 
 .. _`the PyMedPhys GitHub page`: https://github.com/pymedphys/pymedphys
+.. _`examples`: ../user/examples/index.html
+.. _`Installation`: ../getting-started/installation.html
 .. _`Release Notes`: ../getting-started/changelog.html
 .. _`Jupyter notebooks`: https://realpython.com/jupyter-notebook-introduction/
 .. _`continuous integration`: https://en.wikipedia.org/wiki/Continuous_integration
 
 
 .. _source-code:
 
-The PyMedPhys Source Code Package
----------------------------------
+The PyMedPhys Source Code
+-------------------------
 
-All library source code for PyMedPhys is contained within ``src/pymedphys``.
-Within this directory, the code is organised into a range of categories, such
-as ``dicom``, ``gamma``, etc. These correspond to Python modules. Finally, code
-within these categories is organised into levels. Levelling the source code
-helps to prevent circular code dependencies. See diagram below:
+Almost all users will access the PyMedPhys library of source code via the
+main `pymedphys` package (*pymedphys/src/pymedphys/*). However no library code
+actually exists within *pymedphys/src/pymedphys/*. Instead, library code is
+contained within *pymedphys/packages/* and redirected through
+*pymedphys/src/pymedphys/* via a set of python imports.
+
+Within *pymedphys/packages/*, code is organised into a set of subpackages,
+such as ``pymedphys_analysis`` and ``pymedphys_dicom``. From there, each
+subpackage contains a directory named `src/<package_name>`. Within each
```

Comment:
```suggestion
subpackage contains a directory named ``src/<package_name>``. Within each
```

Do we want italics here or code blocks?

Reply:
I struggled to settle on a general convention, to be honest. I started using italics for paths and code blocks for module names (and code, of course), but then broke that rule further down when discussing levelling.

Maybe I'll just lock in code blocks for paths as well.